File: Sources/structlyC/04_linked_list/linked_list_find.cpp
//
//  linked_list_find.h
//  MyCLI
//
//  Created by new on 3/16/26.
//

#include <string>

class Node {
  public:
    std::string val;
    Node *next;

    Node(std::string initialVal) {
        val = initialVal;
        next = nullptr;
    }
};

bool linkedListFind(Node *head, std::string target) {
    Node *node = head;
    while (node != nullptr) {
        if (node->val == target) {
            return true;
        }
        node = node->next;
    }
    return false;
}

bool linkedListFind1(Node *head, std::string target) {
    if (head == nullptr) {
        return false;
    }
    if (head->val == target) {
        return true;
    }
    return linkedListFind(head->next, target);
}

File: Sources/structy/04_linked_list/LinkedListFind.swift
//
//  LinkedListFind.swift
//  MyCLI
//
//  Created by new on 3/16/26.
//

func linkedListFind <T: Equatable>(_ node: Node<T>?, _ target: T) -> Bool {
    guard let node else {
        return false
    }
    return node.val == target || linkedListFind(node.next, target)
}

func linkedListFind1 <T: Equatable>(_ head: Node<T>?, _ target: T) -> Bool {
    var current = head
    while let node = current {
        if node.val == target {
            return true
        }
        current = node.next
    }
    return false
}
